src/Components/WebAssembly/WebAssembly.Authentication/src/Interop/AuthenticationService.ts

```diff
@@ -233,9 +233,11 @@ class OidcAuthorizeService implements AuthorizeService {
     }
 
     private async stateExists(url: string) {
+        const stateHash = new URLSearchParams(new URL(url).hash).get('state');
         const stateParam = new URLSearchParams(new URL(url).search).get('state');
-        if (stateParam && this._userManager.settings.stateStore) {
-            return await this._userManager.settings.stateStore.get(stateParam);
+        const state = stateHash ?? stateParam;
```

Comment:
This lookup needs to be done based on the flow we are doing, it's incorrect to check the hash state if your response mode is query/form post.

Reply:
Ok, makes sense - So far I'm not sure where should I be pulling that response mode information from 😨 Any tips on that?

Reply:
The `ResponseMode` is a setting that users can configure via `OidcProviderOptions`.

https://github.com/dotnet/aspnetcore/blob/master/src/Components/WebAssembly/WebAssembly.Authentication/src/Options/OidcProviderOptions.cs#L58-L59

I believe you can retrive this type from `this_userManager.settings.response_mode`.

Reply:
Thanks @captainsafia - I will take a look into it 😄